app/services/room_settings_getter.rb

```diff
@@ -54,13 +55,17 @@ def infer_specials(room_settings:)
   end
 
   def infer_codes(room_settings:, access_codes:)
-    access_codes.each do |key, code|
-      case room_settings[key]
-      when 'false'
-        room_settings[key] = '' # Forced disbaled access code will have an empty value.
-      when 'true'
-        room_settings[key] = code # Forced enabled access code will conserve its   original value.
-      end
+    filtered_access_codes = access_codes.slice(*room_settings.keys) # Filtering the available room access codes to minimize iterations.
```

Comment:
Access codes when forced disabled and when `only_enabled: true` will be filtered out.
To minimize iteration we ensure that we iterate over the original access codes that do exist on the room setting after filtration. 
Besides it helps simplifies the logic below,